dns_sd_windows.c

```diff
@@ -0,0 +1,262 @@
+// Based on the example provided here: https://github.com/mjansson/mdns
+
+#ifdef _WIN32
+#  define _CRT_SECURE_NO_WARNINGS 1
+#endif
+
+#include <stdio.h>
+#include <errno.h>
+#include "iio-private.h"
+#include "mdns.h"
+#include "network.h"
+#include "debug.h"
+
+#ifdef _WIN32
+#  include <iphlpapi.h>
+#else
+#  include <netdb.h>
+#endif
+
+
+static int new_discovery_data(struct dns_sd_discovery_data** data)
+{
+	struct dns_sd_discovery_data* d;
+
+	d = zalloc(sizeof(struct dns_sd_discovery_data));
+	if (!d)
+		return -ENOMEM;
+
+	*data = d;
+	return 0;
+}
+
+void dnssd_free_discovery_data(struct dns_sd_discovery_data* d)
+{
+	free(d->hostname);
+	free(d);
+}
+
+static int
+open_client_sockets(int* sockets, int max_sockets) {
+	// When sending, each socket can only send to one network interface
+	// Thus we need to open one socket for each interface and address family
+	int num_sockets = 0;
+
+#ifdef _WIN32
+	IP_ADAPTER_ADDRESSES* adapter_address = 0;
+	ULONG address_size = 8000;
+	unsigned int ret;
+	unsigned int num_retries = 4;
+	do {
+		adapter_address = malloc(address_size);
+		ret = GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_SKIP_MULTICAST | GAA_FLAG_SKIP_ANYCAST, 0,
+			adapter_address, &address_size);
+		if (ret == ERROR_BUFFER_OVERFLOW) {
+			free(adapter_address);
+			adapter_address = 0;
+		}
+		else {
+			break;
+		}
+	} while (num_retries-- > 0);
+
+	if (!adapter_address || (ret != NO_ERROR)) {
+		free(adapter_address);
+		IIO_ERROR("Failed to get network adapter addresses\n");
+		return num_sockets;
+	}
+
+	/*int first_ipv4 = 1;
+	int first_ipv6 = 1;*/
```

Comment:
Can remove settings that aren't used, or ifdef them based on `HAVE_IPV6` (which is in iio-config.h)